TG.Backend/Controllers/AuthController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using TG.Backend.Filters;
+
+namespace TG.Backend.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class AuthController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+
+        public AuthController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register([FromBody] AppUserRegisterDTO appUser)
+        {
+            AuthResponseModel resp = await _mediator.Send(new RegisterUserCommand(appUser));
+
+            return resp switch
+            {
+                { IsSuccess: false, StatusCode: HttpStatusCode.BadRequest } => BadRequest(new { resp.Messages }),
+                { IsSuccess: true, StatusCode: HttpStatusCode.OK } => Ok(new { Message = resp.Messages.FirstOrDefault() }),
+                _ => BadRequest()
+            };
+        }
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login([FromBody] AppUserLoginDTO appUser)
+        {
+            AuthResponseModel resp = await _mediator.Send(new LoginUserCommand(appUser));
+
+            return resp switch
+            {
+                { IsSuccess: false, StatusCode: HttpStatusCode.BadRequest } => BadRequest(new { resp.Messages }),
```

Comment:
As we talked, BadRequest is not a good status code, 401 would be better. To fix in other endpoints also :)